src/main/resources/com/google/api/codegen/java/message.snip

```diff
@@ -169,13 +169,6 @@
     @join param : schema.properties
       this.{@param.name} = {@param.name};
     @end
-    ImmutableMap.Builder<String, String> mapBuilder = ImmutableMap.builder();
-    @if schema.hasResourceName
-      {@schema.pathAsResourceName.typeName} resourceName = {@schema.pathAsResourceName.typeName}.parse({@schema.pathAsResourceName.name});
-      @join nameField : schema.pathAsResourceName.fieldCopyMethods
-        mapBuilder.put("{@nameField.name}", resourceName.{@nameField.fieldGetFunction}());
-      @end
-    @end
```

Comment:
Why was this unnecessary?

Reply:
Two things were removed there:
- the mapBuilder that was intended hold the map of field names to their values. This map wasn't ever used in later in the class because I hadn't finished implementing it. If you want, I put a TODO to store the field values in a map instead of in their own separate fields.
- the fields that made up a resource name. Recently, a resource name is just stored as a single String instead of its constituent String parts, and this ApiMessage instance has no idea how to parse a resource name.

Reply:
Sounds good. Storing in separate fields works good I think.